### PR TITLE
Single post UI

### PR DESCRIPTION
**Pull request by sharathnair9999**
Single Post Page with dummy comments listed when clicked on comment button and a new comment box
Profile Page with respective user uploaded posts.
Minor CSS fixes and code refactoring , reduced redundant code and made more reusable

### CODE REVIEW

File: src/components/PostCard/Comment.jsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  const [comment, setComment] = useState(initialCommentState);
+  const [showEmojis, setShowEmojis] = useState(false);
+  const emojisRef = useRef();
+  const [commentedUserInfo, setCommentedUserInfo] = useState({});
+  const [isEditing, setIsEditing] = useState(false);
+  const submitComment = (e) => {
+    e.preventDefault();
+  };
+
+  console.log(existingCommentInfo);
```
Comment (aishwarya-bm) [inline src/components/PostCard/Comment.jsx]: Remove console logs

File: src/pages/UserPosts.jsx [hunk 2]
```diff
@@ -10,7 +16,11 @@ const UserPosts = () => {
   const [userPosts, setUserPosts] = useState([]);
   useEffect(() => {
     const postsRef = collection(db, "posts");
-    const q = query(postsRef, where("uid", "==", profileId));
+    const q = query(
+      postsRef,
+      where("uid", "==", profileId),
```
Comment (aishwarya-bm) [inline src/pages/UserPosts.jsx]: Suggestion to move queries to util files for better modularity

File: src/pages/UserProfile/UserProfile.jsx [hunk 3]
```diff
@@ -39,6 +40,8 @@ const UserProfile = () => {
         joinedAt: getMonthYear(doc.data().joinedAt.seconds),
         link: "",
       });
+      const source = doc.metadata.hasPendingWrites ? "Local" : "Server";
+      console.log(source, " data: ", doc.data());
     });
```
Comment (aishwarya-bm) [inline src/pages/UserProfile/UserProfile.jsx]: Remove console log

**Review by aishwarya-bm — APPROVED** [review]
Overall looks great with clean code and ood component breakdown. I liked the use of useRef hook, window scroll and pure library css without much customisation. Left minor suggestions.

**Review by spraveenofficial — APPROVED** [review]

File: src/pages/UserPosts.jsx (hunk 2 above)
Comment (sharathnair9999) [reply src/pages/UserPosts.jsx]: Done

File: src/components/PostCard/Comment.jsx (hunk 1 above)
Comment (sharathnair9999) [reply src/components/PostCard/Comment.jsx]: Did that

File: src/pages/UserProfile/UserProfile.jsx (hunk 3 above)
Comment (sharathnair9999) [reply src/pages/UserProfile/UserProfile.jsx]: Did it

**Pull request merged**